test/e2e/metrics_test.go

```diff
@@ -0,0 +1,113 @@
+package e2e
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	routev1 "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+	olsv1alpha1 "github.com/openshift/lightspeed-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+var _ = Describe("Prometheus Metrics", Ordered, func() {
+	const metricsViewerSAName = "metrics-viewer-sa"
+	const clusterMonitoringViewClusterRole = "cluster-monitoring-view"
+	var cr *olsv1alpha1.OLSConfig
+	var err error
+	var client *Client
+	var cleanUpFuncs []func()
+	var saToken string
+	var prometheusClient *PrometheusClient
+
+	BeforeAll(func() {
+		client, err = GetClient(nil)
+		Expect(err).NotTo(HaveOccurred())
+		By("Creating a OLSConfig CR")
+		cr, err = generateOLSConfig()
+		Expect(err).NotTo(HaveOccurred())
+		err = client.Create(cr)
+		Expect(err).NotTo(HaveOccurred())
+
+		var cleanUp func()
+
+		By("create a service account")
+		cleanUp, err := client.CreateServiceAccount(OLSNameSpace, metricsViewerSAName)
+		Expect(err).NotTo(HaveOccurred())
+		cleanUpFuncs = append(cleanUpFuncs, cleanUp)
+
+		By("create a role binding for application metrics access")
+		cleanUp, err = client.CreateClusterRoleBinding(OLSNameSpace, metricsViewerSAName, clusterMonitoringViewClusterRole)
+		Expect(err).NotTo(HaveOccurred())
+		cleanUpFuncs = append(cleanUpFuncs, cleanUp)
+
+		By("fetch the service account token")
+		saToken, err = client.GetServiceAccountToken(OLSNameSpace, metricsViewerSAName)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Get a Kubernetes rest config
+		cfg, err := config.GetConfig()
+		Expect(err).NotTo(HaveOccurred())
+
+		openshiftRouteClient, err := routev1.NewForConfig(cfg)
+		Expect(err).NotTo(HaveOccurred())
+
+		prometheusClient, err = NewPrometheusClientFromRoute(
+			context.Background(),
+			openshiftRouteClient,
+			"openshift-monitoring", "thanos-querier",
+			saToken,
+		)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("wait for application server deployment rollout")
+		deployment := &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      OperatorDeploymentName,
+				Namespace: OLSNameSpace,
+			},
+		}
+		err = client.WaitForDeploymentRollout(deployment)
+		Expect(err).NotTo(HaveOccurred())
+
+	})
+
+	AfterAll(func() {
+		for _, cleanUp := range cleanUpFuncs {
+			cleanUp()
+		}
+
+		client, err = GetClient(nil)
+		Expect(err).NotTo(HaveOccurred())
+		By("Deleting the OLSConfig CR")
+		Expect(cr).NotTo(BeNil())
+		err = client.Delete(cr)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should have operator metrics in Prometheus", func() {
+		By("verify Prometheus is scrapping its own metrics")
+		err = prometheusClient.WaitForQueryReturnGreaterEqualOne("prometheus_build_info")
+		Expect(err).NotTo(HaveOccurred())
+
+		By("wait for prometheus scrape metrics from operator")
+		time.Sleep(60 * time.Second)
```

Comment:
better to retry until success (for up to a minute or whatever timeout you want).  Otherwise you're forcing this test to always take at least a minute, even if the metric data might be available sooner.